distribution/src/main/resources/core-cc-ucte-postProcessor.groovy

```diff
@@ -0,0 +1,79 @@
+import com.powsybl.iidm.network.Bus
+import com.powsybl.iidm.network.DanglingLine
+import com.powsybl.iidm.network.EnergySource
+import com.powsybl.iidm.network.LoadType
+import com.powsybl.iidm.network.Network
+import com.powsybl.iidm.network.Network
+import com.powsybl.iidm.network.VoltageLevel
+
+/*
+    UCTE-DEF file does not provide configuration for default nominal voltage setup.
+
+    This post processor script modifies default nominal voltages in order to adapt it to FMAX
+    calculation based on IMAX.
+
+    By default, UCTE sets nominal voltage to 220 and 380kV for the voltage levels 6 and 7, whereas
+    default values of Core countries are 225 and 400 kV instead.
+ */
+
+
+network.voltageLevelStream.forEach({ voltageLevel ->
+    updateVoltageLevelNominalV(voltageLevel)
+})
+
+boolean safeDoubleEquals(double a, double b) {
+    return Math.abs(a - b) < 1e-3
+}
+
+def updateVoltageLevelNominalV(VoltageLevel voltageLevel) {
+    if (safeDoubleEquals(voltageLevel.nominalV, 380)) {
+        voltageLevel.nominalV = 400
+    } else if (safeDoubleEquals(voltageLevel.nominalV, 220)) {
+        voltageLevel.nominalV = 225
+    } else {
+        // Should not be changed cause is not equal to the default nominal voltage of voltage levels 6 or 7
+    }
+}
+
+/*
+    When importing an UCTE network file, powsybl merges its X-nodes into dangling lines.
+
+    It can cause an error if a GLSK file associated to this network includes some factors on
+    xNodes. The GLSK importers looks for a Generator (GSK) or Load (LSK) associated to this
+    xNode. If the Generator/Load does not exist, the GLSK cannot be created.
+
+    This problem has been observed on CORE CC data, on the two Alegro nodes :
+       - XLI_OB1B (AL-BE)
+       - XLI_OB1A (AL-DE)
+
+    This script fix this problem, by creating for these two nodes a fictitious generator (P, Q = 0),
+    connected to the voltage level on which the dangling lines are linked.
+ */
+
+
+createGeneratorOnXnode("XLI_OB1B")
+createGeneratorOnXnode("XLI_OB1A")
+
+void createGeneratorOnXnode(String xnodeId) {
+    Optional<DanglingLine> danglingLine = network.danglingLineStream.filter({dl -> dl.getUcteXnodeCode().equals(xnodeId)}).findAny()
+
+    if (!danglingLine.present) {
+
+        // System.out.println("Xnode " + xnodeId + " not found in network")
```

Comment:
can you remove this line or replace it with a comment ?

Reply:
done, replaced by a comment